### PR TITLE
only make contract unpublished when deployer is in developer mode

### LINKED ISSUE #2683: new flow for deploying evm contract
The developer mode and contract publish thing has introduced a lot of confusion for developers, and it's better to an "opt in" feature when needed, instead of an "extra task" for every developer.

A potential better flow can be something like this
- by default every EOA have developer mode off, and the contracts deployed by them are auto published. This will simplify our current work flow, and align with classic evm flow
- when a developer wants to test a contract before anyone else can interact with it, he can then turn developer mode on for his account. So that every contract by this developer account is unpublished at deployment, and needs manual publish.

### PR DESCRIPTION
**Pull request by xlc**
closes #2683

### CODE REVIEW

File: modules/evm/src/lib.rs [hunk 1]
```diff
@@ -1126,7 +1126,6 @@ pub mod module {
 					}
 
 					if info.exit_reason.is_succeed() {
-						Self::mark_published(contract, Some(source))?;
 						Pallet::<T>::deposit_event(Event::<T>::ContractPublished { contract });
 					}
```
Comment (zjb0807) [inline modules/evm/src/lib.rs:1130]: remove the event

**Review by zjb0807 — COMMENTED** [review]
I feel that the developer status will become useless, no one will use this function, and it may be deprecated in the future.

**Comment by xlc** [comment]
yeah we can check onchain status and see if anyone is using it

File: modules/evm/src/tests.rs [hunk 2]
```diff
@@ -892,8 +874,6 @@ fn create_predeploy_contract_works() {
 			used_storage: 284,
 		}));
 
-		System::assert_last_event(RuntimeEvent::EVM(crate::Event::ContractPublished { contract: addr }));
```
Comment (ermalkaleci) [inline modules/evm/src/tests.rs:895]: are we changing auto publish for predeploy?
https://github.com/AcalaNetwork/Acala/pull/2724/files#diff-db3ad6e2ca083dd40dacda29e393a5d47e24baae25ab8efcab355ed09c5db7efL1128-L1132

**Review by zjb0807 — APPROVED** [review]

File: modules/evm/src/tests.rs (hunk 2 above)
Comment (ermalkaleci) [reply modules/evm/src/tests.rs:895]: oh yes, NetworkContractOrigin is not developer so it will be published

**Review by ermalkaleci — APPROVED** [review]

**Pull request merged**